processInteract rewrites to improve item handling

Tora-B opened the pull request:
I've rewritten a decent chunk of the interaction functions to fix various bugs, behavioral oddities and avoid several potential traps with different kind of fluid handler items. Specifically, healing cows was consuming fluid containers entirely, and not giving the empty container back. I've made a few other changes to cooldown handling and creative mode, as well as tried to make cows immune to damage from their own fluid, since now that healing works correctly, there's a risk of dumping the fluid on the ground when attempting to feed it to a cow.

RoBrit committed:
Update textures filenames to fix missing resources

Tora-B committed:
Only reset cooldown when some interaction succeeds

Tora-B committed:
Ignore cooldown in creative

Tora-B committed:
Rewrite milking function

Attempt to modernize a little to increase compatibility with fluid handlers, and clean up some odd behavior around the edges.

Unlike vanilla cows, fluid cows still give fluid containers in creative.

Tora-B committed:
Rewrite cow healing to prevent containers from being consumed

Only attempt to heal if cow is hurt.
Rewrite item handling to improve compatibility with fluid handlers, and fix items being lost.

Tora-B committed:
Simplify feeding logic

Slight change to align with modern vanilla code.

Tora-B committed:
Make cows immune to damage from their own fluid

As long as the fluid and damage source have the same name, anyway.

Tora-B commented:
Looking through current open issues, this should fix #52.

TheRoBrit commented:
Hi @Tora-B, 

Firstly, apologies for the late response but I've previously had this project on hiatus. I've created an 'experimental' branch for changes suggest by yourself and others, to that end if you would be able to create a new pull-request directed to the 'experimental' branch instead of the '1.12.2' branch, that would be appreciated. 

Thank you.

Tora-B committed:
Guard against wasting breeding items

Normally, the next use cooldown exceeds the Love state, but it could be configured lower.

Tora-B commented:
This pull request includes the changes from #75, and I think I messed up my commits at some point, because they don't appear to apply cleanly. So I'll just close #75 and leave this one open.
